libs/@local/hash-graph-sdk/typescript/src/entity.ts

```diff
@@ -13,9 +17,26 @@ import type {
   EntityProvenance,
   EntityRecordId,
   EntityTemporalVersioningMetadata,
+  EntityUuid,
   LinkData,
 } from "@local/hash-graph-types/entity";
-import { isEqual } from "lodash";
+import type { OwnedById } from "@local/hash-graph-types/web";
+import { isEqual, zip } from "lodash";
+
+import type { AuthenticationContext } from "./authentication-context";
+
+export type CreateEntityParameters = {
+  ownedById: OwnedById;
+  properties: EntityPropertiesObject;
+  linkData?: LinkData;
+  entityTypeId: VersionedUrl;
+  entityUuid?: EntityUuid;
+  draft?: boolean;
+  confidence?: number;
+  propertyMetadata?: PropertyMetadataMap;
+  provenance?: ProvidedEntityEditionProvenance;
+  relationships: EntityRelationAndSubject[];
+};
```

Comment:
I think @CiaranMn already pointed this out, but it'd be nice to re-use the GraphApi type for the create entity request here and only override the types that don't correspond. That way in future if we add additional fields, we don't have to add them here by default. I think it'd become something like this:
```suggestion
export type CreateEntityParameters = {
  ownedById: OwnedById;
  properties: EntityPropertiesObject;
  entityTypeId: VersionedUrl;
  entityUuid?: EntityUuid;
  draft?: boolean;
} & Omit<
  CreateEntityRequest,
  "ownedById" | "properties" | "entityTypeId" | "entityUuid" | "draft"
>;
```

Reply:
Yes, I actually just prepared a commit for this:
```ts
export type CreateEntityParameters = Omit<
  GraphApiCreateEntityRequest,
  "entityTypeIds" | "decisionTime"
> & {
  ownedById: OwnedById;
  properties: EntityPropertiesObject;
  linkData?: LinkData;
  entityTypeId: VersionedUrl;
  entityUuid?: EntityUuid;
  propertyMetadata?: PropertyMetadataMap;
};

export type PatchEntityParameters = Omit<
  GraphApiPatchEntityParams,
  "entityId" | "entityTypeIds" | "decisionTime"
> & {
  entityTypeId?: VersionedUrl;
};
```